### PR TITLE
Table layout improvements

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves most of #4621 ("Up our table layout game") and a couple things in #4913 ("Drill Polish 3"):

1. tighten up row height #4621
2. right align numbers #4621
3. don't show sort arrow on hover (only if column is sorted) #4913
4. make header click target full cell #4913

### CODE REVIEW

**Comment by tlrobinson** [comment]
Also I haven't implemented these changes for the dashboard version of tables yet. If this looks good I'll do that.

File: frontend/src/metabase/lib/formatting.js [hunk 1]
```diff
@@ -64,20 +78,54 @@ function formatMajorMinor(major, minor, options = {}) {
...
+export function formatTimeRangeWithUnit(value: Value, unit: DatetimeUnit, options: FormattingOptions = {}) {
     let m = parseTimestamp(value, unit);
     if (!m.isValid()) {
         return String(value);
     }
+
+    // Tooltips should show full month name, but condense "MMMM D, YYYY - MMMM D, YYYY" to "MMMM D - D, YYYY" etc
+    const monthFormat = options.type === "tooltip" ? "MMMM" : "MMM";
+    const condensed = options.type === "tooltip";
+    // use en dashes, for Maz
```
Comment (attekei) [inline frontend/src/metabase/lib/formatting.js]: 😄

**Review by attekei — COMMENTED** [review]
I really like the new table layout! My only concern is that now scrolling the table is less smooth than before. Do you know if it's easy to improve the scrolling performance?

I also tested that the new week numbering seems to work well.

File: frontend/src/metabase/lib/formatting.js (hunk 1 above)
Comment (tlrobinson) [inline frontend/src/metabase/lib/formatting.js]: Hmm it does add an additional DOM node per cell, that's the only reason I can think of it being less smooth. That seemed necessary to get both the contents aligned vertically, and the ellipsis behavior.

We don't need to use flex to align it vertically, it just seemed cleaner than picking magic padding sizes.

**Comment by tlrobinson** [comment]
Hmm it does add an additional DOM node per cell, that's the only reason I can think of it being less smooth. That seemed necessary to get both the contents aligned vertically, and the ellipsis behavior.

We don't need to use flex to align it vertically, it just seemed cleaner than picking magic padding sizes.

**Comment by attekei** [comment]
I think that the difference is smoothness is mostly caused by higher density of table cells, so probably the extra DOM node per cell isn't such a big factor.

**Review by kdoh — APPROVED** [review]

**Comment by tlrobinson** [comment]
@metabase/core-developers mind trying this out and letting me know if you notice any worse scroll performace?

**Review by mazameli — APPROVED** [review]
LGTM!

**Comment by salsakran** [comment]
I'm still seeing the sort icon on hover.

Also, definitely getting more lag on scrolling.

**Comment by tlrobinson** [comment]
@attekei @salsakran Do you mind reverting just the ROW_HEIGHT constant (https://github.com/metabase/metabase/pull/4977/files#diff-fb164def68c92e5830ecc8ac0f165a2aR25) and seeing if that fixes the scroll lagginess for you? That will tell us if it's the additional DOM nodes or additional row density.

**Comment by tlrobinson** [comment]
@kdoh @attekei and ideas how to avoid introducing the extra DOM node and still use flex align-center + ellipsification?

**Comment by attekei** [comment]
My first idea is to get rid of that `<span>` in https://github.com/metabase/metabase/blob/6f13cbdbcfbabfcfcb7786589a284e988778fc1a/frontend/src/metabase/components/Value.jsx#L19, what do you think? It could already be a considerable perf boost.

**Comment by tlrobinson** [comment]
That's there because `formatValue` might return a string which isn't a valid React element.

I suppose we could just use `formatValue` directly instead of the `Value` component.

**Comment by tlrobinson** [comment]
I can't repro "I'm still seeing the sort icon on hover.", can anyone else?

**Comment by tlrobinson** [comment]
Ok I removed the extraneous <span> and it feels slightly faster, but I do think the majority of the performance regression is just due to the higher density of cells.

I've also implemented right alignment on the dashboard version of tables, so I think this is ready for review.

**Review by attekei — COMMENTED** [review]
Yep, it feels faster (hopefully it's not all placebo 😉), but I'm not sure if the current density is good or if the row height should be increased a little.

One other minor question: should Lat/Lon values also be right-aligned?

**Comment by tlrobinson** [comment]
I agree it feels slightly too dense. I've increased the row height from 28px:

screenshot 2017-05-19 14 06 35

to 30px:

screenshot 2017-05-19 14 06 16

I've also included lat/lon in the right aligned fields.

**Comment by tlrobinson** [comment]
This should be ready to go, or at least another round of review.

**Comment by salsakran** [comment]
feels reasonable to me.

at this point the grid still irritates me, but right aligning stuff is really nice.

**Comment by mazameli** [comment]
As they say in the freight industry, "ship it."

**Pull request merged**